scripts/ci-build.sh

```diff
@@ -0,0 +1,26 @@
+#!/bin/bash
+
+# Copyright 2018 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+REPO_ROOT=$(dirname "${BASH_SOURCE}")/..
+
+go build \
+   "${REPO_ROOT}"/cluster-api/... \
+   "${REPO_ROOT}"/cluster-api-gcp/... \
```

Comment:
(sanity check)

Have you tried this command to validate that is actually recursively builds? ie. custer-api-gcp/machine-controller builds.

I tried to manually run go build full-to-cluster-api-gcp/... and I got warnings that it was not a package. Could just be me though.

Reply:
I suspect you were in the repo root and ran something like `go build cluster-api-gcp/...`. Try `go build ./cluster-api-gcp/...`. The relative path is important, otherwise it will try to compile `$GOPATH/src/cluster-api-gcp/...` which it won't find.

Reply:
You are right. That and I was not running go1.9